### PR TITLE
Version with lots of improvements

### PR DESCRIPTION
**Pull request by Sylvain78**
Improvements.

Tested on x86_gcc2 and x86 ,bare metal installation

### CODE REVIEW

File: Src/Headers/TButton.h [hunk 1]
```diff
@@ -11,11 +11,12 @@
 #include <File.h>
 #include <String.h>
 #include <vector>
+using namespace std;
```
Comment (waddlesplash) [inline Src/Headers/TButton.h:14]: It would be better to use `using std::vector`.

File: Src/Headers/TexBar.h [hunk 2]
```diff
@@ -11,9 +11,11 @@
 #include <ScrollBar.h>
 #include <InterfaceDefs.h>
 #include <vector>
+using namespace std;
```
Comment (waddlesplash) [inline Src/Headers/TexBar.h:14]: same here

**Comment by scottmc** [comment]
@Sylvain78 , 
While you are updating this, you can decompile the rsrc file using 'rc' and turn it into an rdef file.  
See: https://www.haiku-os.org/documents/dev/compile_them_resources - instructions for decompiling are at the end of the article.  It's a good read and helps explain why we prefer to have rdef over rsrc files in git.  Then you just update the makefile and it should take care of it for you.  See other makefiles on haikuarchives that have rdef files present for examples.

Also see #14 there is a source zip file on the linked page there.  To me it looks like an earlier rev.  But the poster say that version is more stable.  You might check it to see if there is anything on use in that source zip.  If you check it over, you might comment on that issue regarding what you find and maybe we can close that one out.

**Comment by scottmc** [comment]
Also, I see you did your work in master.  We prefer you follow the HaikuPorts/HaikuArchives development model:
https://github.com/haikuports/haikuports/wiki/DevelopmentModel

Don't worry about it for this PR, but once this gets merged, start doing future work in feature branches.  After you do a few Pull Requests following this model you will get the hang of how it works.

**Comment by waddlesplash** [comment]
> Also, I see you did your work in master. We prefer you follow the HaikuPorts/HaikuArchives development model:

Ultimately it does not make much of a difference what branch a PR comes from for us; it just makes it easier on the person sending PRs to reset state once the PR has been merged.

File: Makefile [hunk 3]
```diff
@@ -28,49 +29,51 @@ APP_MIME_SIG =
...
+ Src/Source/ColorWindow.cpp \
+ Src/Source/Preferences.cpp \
+ Src/Source/Prefs.cpp \
+ Src/Source/PrefsListItem.cpp \
+ Src/Source/PrefsListView.cpp \
+ Src/Source/PrefsWindow.cpp
 
 #	Specify the resource definition files to use. Full or relative paths can be
 #	used.
 RDEFS =
```
Comment (scottmc) [inline Makefile:72]: Switch to use the rdef instead of rsrc"

RDEFS = BeTeX.rdef

RSRC =

**Comment by Begasus** [comment]
Any progress here? With the open PR at haikuports (and the above fix for 32bit) it builds OK and works fine, current version doesn't let me open tex files, with these changes (and the work done on texlive/latex2html) it works quite well.

**Comment by scottmc** [comment]
What's the status on this one?  Do these changes still work, what is left to be done?  Is this ready for merge?

**Comment by Begasus** [comment]
Patch doesn't apply, getting fails on Makefile (did a srcGitRev with latest commit and this PR as a patch)

**Pull request closed without merging**